app/Http/Controllers/HomeController.php

```diff
@@ -99,16 +102,49 @@ public function getChangelog()
             ->get();
 
         $featuredStream = null;
+        $activeStream = null;
 
         foreach ($streams as $index => $stream) {
+            if ($stream->version === $buildId) {
+                $activeStream = $stream->stream_id;
+            }
+
             if ($stream->stream_id === config('osu.changelog.featured_stream')) {
                 $featuredStream = $stream;
                 unset($streams[$index]);
-                break;
             }
         }
 
-        return view('home.changelog', compact('changelogs', 'streams', 'featuredStream', 'build'));
+        $buildsTable = with(new Build)->getTable();
+        $propagationTable = with(new BuildPropagationHistory)->getTable();
+        $buildHistory = BuildPropagationHistory::changelog($activeStream, config('osu.changelog.chart_days'))
+            ->get()
+            ->map(function ($item) {
+                $item->user_count = get_int($item->user_count);
+
+                return $item;
+            });
+
+        $chartOrder = null;
+
+        if ($buildId === null) {
```

Comment:
may as well make a dedicated controller instead of keep mixing between index and show in single action.